nordicsemi/zigbee/ota_file.py

```diff
@@ -74,25 +78,45 @@ def __init__(self,
                  firmware,
                  manufacturer_code = OTA_UPGRADE_MANUFACTURER_WILDCARD,
                  image_type = OTA_UPGRADE_IMAGE_TYPE_WILDCARD,
-                 comment = ''):
+                 comment = '',
+                 min_hw_version=None,
+                 max_hw_version=None):
         '''A constructor for the OTA file class, see Zigbee ZCL spec 11.4.2 (Zigbee Document 07-5123-06)
            see: http://www.zigbee.org/~zigbeeor/wp-content/uploads/2014/10/07-5123-06-zigbee-cluster-library-specification.pdf
            (access verified as of 2018-08-06)
         '''
+
         total_len = OTA_UPGRADE_FILE_HEADER_LENGTH + 3 * OTA_UPGRADE_SUBELEMENT_HEADER_SIZE + OTA_UPGRADE_SUBELEMENT_TRIGGER_LENGTH + init_cmd_len + firmware_len
+
         ota_header_pack_format = '<LHHHHHLHc31sL'
-        ota_header = struct.pack(ota_header_pack_format,
-                                 OTA_UPGRADE_FILE_HEADER_FILE_ID,
-                                 OTA_UPGRADE_FILE_HEADER_FILE_VERSION,
-                                 OTA_UPGRADE_FILE_HEADER_LENGTH,
-                                 OTA_UPGRADE_FIELD_CONTROL,
-                                 manufacturer_code,
-                                 image_type,
-                                 file_version,
-                                 OTA_UPGRADE_FILE_HEADER_STACK_PRO,
-                                 chr(len(comment)),
-                                 bytes(comment.encode('ascii')),
-                                 total_len)
+        ota_header_pack_args = [OTA_UPGRADE_FILE_HEADER_FILE_ID,
+                                OTA_UPGRADE_FILE_HEADER_FILE_VERSION,
+                                OTA_UPGRADE_FILE_HEADER_LENGTH,
+                                OTA_UPGRADE_FIELD_CONTROL,
+                                manufacturer_code,
+                                image_type,
+                                file_version,
+                                OTA_UPGRADE_FILE_HEADER_STACK_PRO,
+                                chr(len(comment)),
+                                bytes(comment.encode('ascii')),
+                                total_len]
+
+        if (type(min_hw_version) is int) and (type(max_hw_version) is int):
+            # Calculate additional length of minimum and maximum hardware version fields
+            ota_min_max_hw_length = OTA_UPGRADE_MIN_HW_VERSION_LENGTH + OTA_UPGRADE_MAX_HW_VERSION_LENGTH
+            # Add additional bytes to formating string
+            ota_header_pack_format += 'HH'
```

Comment:
Is there any check for values passed as min/max, so each of them fit in 8-bit field?